drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -182,6 +182,12 @@ static void scan_done_handler(void)
 			strncpy(res.ssid, ap_list_buffer[k].ssid, ssid_len);
 			res.rssi = ap_list_buffer[k].rssi;
 			res.channel = ap_list_buffer[k].primary;
+
+			if (ap_list_buffer[k].bssid != NULL) {
```

Comment:
`ap_list_buffer[k].bssid` will never be NULL. Just remove the if clause and it shall work. Worst scenario is empty bssid.